lib/samson/integration.rb

```diff
@@ -0,0 +1,16 @@
+module Samson
+  class Integration
+
+    @@sources = Rails.root.join('app','controllers', 'integrations').children(false).map do |controller_path|
+      controller_path.to_s[/\A(?!base)(\w+)_controller.rb\z/, 1]
+    end.compact
+
+    cattr_reader :sources
+
+    def self.register_source(source)
```

Comment:
not used <-> don't add ? ... maybe also just `Integration.sources << xxx` is a good enough api :)